### PR TITLE
Implement Janrain Capture and Janrain Engage auth provider

### PR DESCRIPTION
**Pull request by AndrewLane**
Fixes https://github.com/ParsePlatform/parse-server/issues/2398

An addition to parse server to allow for authentication using Janrain. Janrain provides an Engage product as well as a Capture product, so there will be separate auth providers for each. It would be similar to Twitter, Facebook, etc. which are already implemented.

**1) Janrain Engage Configuration:**

The Janrain Engage API Key can be provided to Parse Server via configuration like:

```
{
    "appId": "...",
    "masterKey": "...",
    "databaseURI": "...",
    /* ... */
    "oauth": {
        "janrainengage": {
            "api_key": "your_api_key_here"
        }            
    }
}
```

**2) Janrain Engage Auth Request**

The POST to ```/users``` looks like:

```
{
    "authData": {
        "janrainengage": {
            "id": "social_id_here",
            "auth_token": "janrain_engage_auth_token_here"
        }
    }
}
```

**3) Janrain Capture Configuration:**

The Janrain Capture host can be provided to Parse Server via configuration like:

```
{
    "appId": "...",
    "masterKey": "...",
    "databaseURI": "...",
    /* ... */
    "oauth": {
        "janraincapture": {
            "janrain_capture_host": "your-janrain-capture-host.janraincapture.com"
        }            
    }
}
```

**4) Janrain Capture Auth Request**

The POST to ```/users``` looks like:

```
{
    "authData": {
        "janraincapture": {
            "id": "janrain_capture_id",
            "access_token": "janrain_capture_access_token"
        }
    }
}
```

### CODE REVIEW

**Comment by AndrewLane** [comment]
Whoops, I added a breaking unit test.  Sorry for the amateur hour.  Will get that fixed.

**Comment by facebook-github-bot-9** [comment]
@AndrewLane updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@AndrewLane updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@AndrewLane updated the pull request.

File: src/authDataManager/janrainengage.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    post_req.write(post_data);
+    post_req.end();
+  });
+}
+
+module.exports = {
+  validateAppId: validateAppId,
+  validateAuthData: validateAuthData
+};
```
Comment (flovilmart) [inline src/authDataManager/janrainengage.js]: add line breack

Comment (AndrewLane) [inline src/authDataManager/janrainengage.js]: just to clarify, the standard line break at the end of the file needs to be added?

Comment (flovilmart) [inline src/authDataManager/janrainengage.js]: yes

**Comment by facebook-github-bot-5** [comment]
@AndrewLane updated the pull request.

File: src/authDataManager/janrainengage.js (hunk 1 above)
Comment (AndrewLane) [inline src/authDataManager/janrainengage.js]: ok, just added https://github.com/AndrewLane/parse-server/commit/b0fe257f0ad4dd0c35cfc4aa93362a3f236bddb6

Comment (AndrewLane) [inline src/authDataManager/janrainengage.js]: should i squash commits?

**Comment by flovilmart** [comment]
They are gonna Be squashed when merging

**Comment by AndrewLane** [comment]
:+1:

**Pull request merged**